Updated durable function orchestrator example to .NET 8

charloconnell opened the pull request:
###Overview: Updated durable function orchestrator example to .NET 8, this involved updating the function app to follow the isolated worker model for function apps (as opposed to the in-process model that was being used previously). For more information see: https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-in-process-differences

Updated the unit tests and documentation to reflect these changes as well

charloconnell committed:
Converted to isolated worker model

Converted example orchestrator function to run using the isolated worker model for function apps.

charloconnell committed:
Updated unit tests

Updated unit tests to cater for changes from moving orchestrator to isolated worker model

charloconnell committed:
Updated all nuget packages to latest version

charloconnell committed:
Upgraded to .NET 8

Projects upgraded:
-Orchestrator function app
-Unit test project

charloconnell committed:
Update README.md

Updated readme to reflect the .NET 8 and isolated worker model changes

micheal-wheeler-1nebula commented on DurableFunctionOrchestratorExample/DurableFunctionOrchestrator.cs:
```diff
@@ -13,29 +14,31 @@ namespace DurableFunctionOrchestratorExample
+            [DurableClient] DurableTaskClient durableOrchestrationClient)
         {
-            log.LogInformation($"C# ServiceBus queue trigger function processed message: {myQueueItem}");
+            _logger.LogInformation($"C# ServiceBus queue trigger function processed message: {myQueueItem}");//TODO serialize
```
Please implement this `TODO`

micheal-wheeler-1nebula commented on Stactize.DurableFunctionOrchestratorExample.Tests/DurableOrchestratorTests.cs:
```diff
@@ -144,18 +145,18 @@ public async Task RunOrchestratorWithEventSuspend_Should_CallActivity()
             context.Setup(x => x.GetInput<OrchestrationActionModel>())
                    .Returns(knownActionModel);
-            context.Setup(x => x.CallActivityAsync<OrchestrationResultModel>(Constants.DurableActivity.Suspend, knownActionModel))
+            context.Setup(x => x.CallActivityAsync<OrchestrationResultModel>(Constants.DurableActivity.Suspend, knownActionModel, null))
```
I think using `It.IsAny<TaskOptions>()` as a parameter instead of null for these Setup and Verify calls is more robust in case any options are added in the future.

charloconnell committed:
Implemented pull request feedback

- Updated unit test method set-ups to be more future-proof
- Updated logged object in orchestrator to be serialized before logging
- Updated orchestrator logs to be tempate-based
- Updated system.text serialization options to handle integer enums as well as string

charloconnell committed:
Updated orchestrator to handle messages with comments

Updated json serializer options to ignore comments in json strings since our example json payload has comments in it
